src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -188,6 +201,89 @@ template<class T> bool boxFrustum(const Range3D<T>& box, const Frustum<T>& frust
     return true;
 }
 
+template<class T> bool sphereFrustum(const Vector3<T>& center, const T radius, const Frustum<T>& frustum) {
+    const T radiusSq = radius*radius;
+
+    for(const Vector4<T>& plane: frustum.planes()) {
+        /* The sphere is in front of one of the frustum planes (normals point
+           outwards) */
+        if(Distance::pointPlaneScaled<T>(center, plane) < -radiusSq)
+            return false;
+    }
+
+    return true;
+}
+
+/**
+@brief Intersection of a point and a cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param angle    Apex angle of the cone
+
+Returns `true` if the point is inside the cone.
+
+@see pointCone(Vector3, Vector3, Vector3, T) @todo: That ref is never going to work...
+*/
+template<class T> bool pointCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, const Deg<T> angle) {
+    const T x = T(1)+Math::pow<T>(Math::tan<T>(angle/T(2)), T(2));
+
+    return pointCone(p, origin, normal, x);
+}
+
+/**
+@brief Intersection of a point and a double cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param angle    Apex angle of the cone
+
+Returns `true` if the point is inside the double cone.
+
+@see pointDoubleCone(Vector3, Vector3, Vector3, T) @todo: That ref is never going to work...
+*/
+template<class T> bool pointDoubleCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, const Deg<T> angle) {
+    const T x = T(1)+Math::pow<T>(Math::tan<T>(angle/T(2)), T(2));
+
+    return pointDoubleCone(p, origin, normal, x);
+}
+
+/**
+@brief Faster version of intersection of a point and a cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param tanAngleSqPlusOne Precomputed portion of the cone intersection equation: `Math::tan(angle/2)+1`
+
+Returns `true` if the point is inside the cone.
+
+Uses the result of precomputing @f$x = \tan^2{\theta} + 1@f$.
+*/
+template<class T> bool pointCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, const T tanAngleSqPlusOne) {
+    const Vector3<T> c = p - origin;
+    const T lenA = dot(c, normal);
+
+    return lenA >= 0 && c.dot() <= lenA*lenA*tanAngleSqPlusOne;
+}
+
+/**
+@brief Faster version of intersection of a point and a double cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param tanAngleSqPlusOne Precomputed portion of the cone intersection equation: `Math::tan(angle/2)+1`
+
+Returns `true` if the point is inside the double cone.
```

Comment:
I'm such a crazy person that I usually highlight even this: `@cpp true @ce` (it'll make it appear bold, in case you are wondering).

You are free to say "no" to this, if such extra work offends you :D

Reply:
As I said often times before... I use vim 😈